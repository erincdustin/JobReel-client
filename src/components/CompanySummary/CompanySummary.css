.saved-company-list {
    position: relative;
    grid-column-start: 4;
    grid-column-end: 9;
    height: 500px;
    border-radius: 10px;
    position: relative;
    text-align: center;
    overflow-y: auto;
}

.savedCompanyFilterControls{
    width: 90%;
    text-align: right;
    margin: 15px;
}

@media screen and (max-width: 960px) {
    .saved-company-list {
        grid-column-start: 1;
        grid-column-end: 9;
    }
}