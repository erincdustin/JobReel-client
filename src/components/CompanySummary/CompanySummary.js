import React from 'react';
import Button from '../../components/Button/Button';
import JobReelContext from '../../context/JobReelContext';
import jobReelApiService from '../../services/jobreel-api-service';
import Company from '../Company/Company';
import './CompanySummary.css';

class CompanySummary extends React.Component {

  static contextType = JobReelContext;

  handleClickDelete(companyId){
    jobReelApiService.deleteCompany(companyId)
    this.context.deleteCompany(companyId)
  }

  render(){
    let mappedCompanies;
    if(this.context.savedCompanies !== []){
      let companies = this.context.companies
      mappedCompanies = companies
      .map(company => <Company id={company.company_id} name={company.company_name} city={company.city} state={company.state} date={company.date_added} industry={company.industry} desc={company.description} website={company.website} contact={company.contact} user={company.user_id}/>)
    }

    return(
      <div className="saved-company-list">
        <div className='saved-company-controls'>
          <Button onClick={() => this.context.setManualCompanyAdd(true)} type="button">Add Company</Button>
        </div>
        
        {mappedCompanies}
      </div>
    )
  }
}

export default CompanySummary