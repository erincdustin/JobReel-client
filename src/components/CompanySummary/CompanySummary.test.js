import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import CompanySummary from './CompanySummary'

describe('CompanySummary smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CompanySummary />, div);
    ReactDOM.unmountComponentAtNode(div);
  })
})

describe('CompanySummary snapshot test', () => {
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<CompanySummary />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})