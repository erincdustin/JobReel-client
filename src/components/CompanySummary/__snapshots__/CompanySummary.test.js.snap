// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanySummary snapshot test renders the UI as expected 1`] = `
<div
  className="saved-company-list"
>
  <h2>
    Saved Companies
  </h2>
  <button
    className="Button"
    onClick={[Function]}
    type="button"
  >
    Add Company
  </button>
</div>
`;
